src/ripple/consensus/Consensus.cpp

```diff
@@ -49,23 +64,34 @@ shouldCloseLedger(
         return true;
     }
 
-    if ((proposersClosed + proposersValidated) > (prevProposers / 2))
+    if (prevProposers >= 1)
     {
-        // If more than half of the network has closed, we close
-        JLOG(j.trace()) << "Others have closed";
-        return true;
+        if ((proposersClosed + proposersValidated) > (prevProposers / 2))
+        {
+            // If more than half of the network has closed, we close
+            JLOG(j.trace()) << "Others have closed"
+                            <<  " proposersClosed:  " << proposersClosed
+                            <<  " proposersValidated: " << proposersValidated;
+            return true;
+        }
     }
 
     if (!anyTransactions)
     {
-        // Only close at the end of the idle interval
-        return timeSincePrevClose >= idleInterval;  // normal idle
+        // Only close at the end of the idle interval when no txs
+        JLOG(j.debug()) << "Checking idle interval (no txs)"
+                        << " timeSincePrevClose: " << timeSincePrevClose.count()
+                        << " minOpenTime: " << minOpenTime.count()
+                        << " idleInterval: " << idleInterval.count();
+        return timeSincePrevClose >= std::max(minOpenTime,idleInterval);
```

Comment:
Also not new, but why can't we get away with just using `openTime` here instead of `timeSincePrevClose`?
It would allow us to remove this, which is sometimes comparing `now_` with our precise ledger close time and other times comparing with the consensus time:
https://github.com/bachase/rippled/blob/open-delay/src/ripple/consensus/Consensus.h#L1090-L1107

Reply:
We might be able to. The main differences are:
1. If actually accepting the ledger takes a long time, `openTime` doesn't account for that but `timeSincePrevClose` does.
2. If you switch ledgers during a consensus round, `openTime` would also reset. Arguably you would close anyway because your peers are probably ahead of you.

Regardless, this is only when you have no transactions in the open ledger, which is a big of an edge case anyway. My instinct is to stick with what we have, just to limit the number of changes. `timeSincePrevClose` is used here https://github.com/ripple/rippled/pull/2548/files/40993bf8e6faf3c7cb0495b19c846096530c8632#diff-caed2749d18cdffcee64059060c1c14dR56, although we could also consider removing that too.